spec/preloading_spec.cr

```diff
@@ -245,4 +270,30 @@ describe "Preloading" do
 
     posts.results.first.comments.should eq([comment])
   end
+
+  describe "when there's no results in the parent query" do
+    it "skips running the preload query for has_many" do
+      Comment::BaseQuery.times_called = 0
+      posts = Post::BaseQuery.new.preload_comments
+      posts.results
+
+      Comment::BaseQuery.times_called.should eq 0
```

Comment:
One way to make sure this spy is working is to add a couple of these counts in other spots to see that it is calling it only once! This would be an awesome spce to have to make sure preloading is not accidentally calling things twice :D